### ISSUE TITLE
[Enhancement] RaiseAllPropertiesChanged(); on BindableBase.

### ISSUE THREAD

**Issue by vpenades** [PrismLibrary/Prism#2609]
## Summary

It is fairly common to call `RaisePropertyChanged(null)` to update all the properties in one shot.

But the current definition of `RaisePropertyChanged` which uses `[CallerMemberName]` interferes with that behavior.

The alternative is to call:

```c#
OnPropertyChanged(new System.ComponentModel.PropertyChangedEventArgs(null));
```

## API Changes

I would like to add this method to MVVM.BindableBase:

```c#
private static readonly AllProperties = new System.ComponentModel.PropertyChangedEventArgs(null);

public void RaiseAllPropertiesChanged()
{
   OnPropertyChanged( AllProperties );
}

```

## Intended Use Case

`RaiseAllPropertiesChanged()` would be used as:
- The short version of `OnPropertyChanged(new System.ComponentModel.PropertyChangedEventArgs(null));` 
- Disambiguation with `RaisePropertyChanged(null);`

If agreed I can create a PR for this change.

**Comment by brianlagunas** [comment]
`RaisePropertyChanged(null);` works as expected.

**Comment by vpenades** [comment]
> `RaisePropertyChanged(null);` works as expected.

No, it does not.

See https://github.com/dotnet/roslyn/issues/37152

Apparently, when you pass null as an argument that has been defined with [CallerMemberName] , the null value is _replaced_ by the actual caller member name string. Which is not the intended use case.

So right now, it is not possible to raise "all properties" changed with the RaisePropertyChanged method.

**Comment by brianlagunas** [comment]
I just ran a sample and it worked. Can you can provide a sample that shows it doesn't?

**Comment by vpenades** [comment]
I've just wrote a fast test and I got more insight of what's going on...

As it happens, `RaisePropertyChanged(null);` but `RaisePropertyChanged();` does not.

Since RaisePropertyChanged is defined as:

```c#
void RaisePropertyChanged([CallerMemberName] string propertyName = null);
```

you can think that both:

```c#
RaisePropertyChanged(null);
RaisePropertyChanged();
```
do exactly the same.... but in fact they behave in a completely different way.

Also one of the code analyzers we use, recomends removing the "null", because since propertyName is defined with the default null value, which we did, and that's where we began noticing that the bindings stopped working.

Now I understand that in order to raise for "all properties" I have to explicitly state the value "null", even if the propertyName default value is already set to null, because [CallerMemberName] is doing a mess with the default value, which is something very counterintuitive.... and that's why I proposed the API change.

**Comment by brianlagunas** [comment]
Correct. This feature is working exactly how it was designed by MS. Calling `RaisePropertyChanged()` will use the CallerMemberName attribute which will provide the name of the property as a string. Calling `RaisePropertyChanged(null)` will cause all properties to notify.

**Comment by vpenades** [comment]
so it's a case of being a feature, not a bug....

then, well closed it is.
